native/src/routes/SearchModal.tsx

```diff
@@ -56,45 +55,37 @@ class SearchModal extends React.Component<PropsType, SearchStateType> {
   findCategories(categories: CategoriesMapModel): Array<ListEntryType> {
     const { query } = this.state
     const normalizedFilter = normalizeSearchString(query)
-    const categoriesArray = categories.toArray()
-    // find all categories whose titles include the filter text and sort them lexicographically
+    const categoriesArray = categories.toArray().filter(it => !it.isRoot())
+
+    // Lexicographically sorted categories with match in title
     const categoriesWithTitle = categoriesArray
-      .filter(category => normalizeSearchString(category.title).includes(normalizedFilter) && !category.isRoot())
-      .map(
-        (category: CategoryModel): ListEntryType => ({
-          model: {
-            title: category.title,
-            thumbnail: category.thumbnail,
-            path: category.path,
-            contentWithoutHtml: parseHTML(category.content),
-            titleMatch: true,
-          },
-          subCategories: [],
-        })
-      )
+      .filter(category => normalizeSearchString(category.title).includes(normalizedFilter))
+      .map(category => ({
```

Comment:
Types are unnecessary here because they are implicitly asserted by typescript. It makes things easier readable and removes the necessity to check for `contentWithoutHtml` later on compared to keeping the type `ListEntryType`.